### PR TITLE
feat: wasm

### LINKED ISSUE #63: Wasm Build
Let's check for wasm build early so we don't get caught up with technical debt when we start shipping wasm.

See https://github.com/web-infra-dev/oxc/blob/3605f7c0fc2421496a869c2b8fb470340903a21f/.github/workflows/ci.yml#L57

### PR DESCRIPTION
**Pull request by houyunlu**
### Description
1. Closed https://github.com/rolldown-rs/rolldown/issues/63
2. FInished wasm32 target compilation

### Test Plan

---

### CODE REVIEW

File: crates/rolldown-wasm/Cargo.toml
Comment (hyf0) [inline crates/rolldown-wasm/Cargo.toml:1]: should be underscope.

Comment (houyunlu) [reply crates/rolldown-wasm/Cargo.toml:1]: ok

File: crates/rolldown_utils/src/lib.rs [hunk 1]
```diff
@@ -1,20 +1,5 @@
 mod magic_string_ext;
-use std::future::Future;
 
-use async_scoped::TokioScope;
 pub mod reserved_word;
 
-pub fn block_on_spawn_all<Iter, Out>(iter: Iter) -> Vec<Out>
```
Comment (houyunlu) [inline crates/rolldown_utils/src/lib.rs:7]: explained before.

File: crates/rolldown/src/bundler/module_loader/module_loader.rs [hunk 2]
```diff
@@ -113,11 +112,11 @@ impl<'a, T: FileSystemExt + 'static> ModuleLoader<'a, T> {
   }
 
   #[allow(clippy::collection_is_never_read)]
-  fn resolve_entries(&mut self) -> BatchedResult<Vec<(Option<String>, ResolvedRequestInfo)>> {
+  async fn resolve_entries(&mut self) -> BatchedResult<Vec<(Option<String>, ResolvedRequestInfo)>> {
     let resolver = &self.resolver;
 
     let resolved_ids =
-      block_on_spawn_all(self.input_options.input.iter().map(|input_item| async move {
+      futures::future::join_all(self.input_options.input.iter().map(|input_item| async move {
```
Comment (houyunlu) [inline crates/rolldown/src/bundler/module_loader/module_loader.rs:197]: Use `futures::join_all` to avoid introducing another crate, they have same effect.

**Comment by hyf0** [comment]
Would you mind on hold this pr before https://github.com/rolldown-rs/rolldown/pull/182 being merged?

File: crates/rolldown/src/bundler/module_loader/module_loader.rs (hunk 2 above)
Comment (hyf0) [reply crates/rolldown/src/bundler/module_loader/module_loader.rs:197]: I'm not sure. `block_on_spawn_all` would run tasks in parallel while `join_all` seems to run them concurrently.

**Comment by houyunlu** [comment]
!bench

**Review by hyf0 — APPROVED** [review]

**Pull request merged**